src/components/container/BloodPressureVisualization/BloodPressureVisualization.tsx

```diff
@@ -0,0 +1,303 @@
+import "./BloodPressureVisualization.css";
+import { useEffect, useState } from "react";
+import MyDataHelps from "@careevolution/mydatahelps-js";
+import React from "react";
+import { Button, DateRangeNavigator, LoadingIndicator, Title } from "../../presentational";
+import DumbbellChart from "../../presentational/DumbbellChart";
+import { Dumbbell, ClosedInterval, Axis, DataPoint, DumbbellClass } from "../../presentational/DumbbellChart/DumbbellChart";
+import { addDays, format, isEqual, isMonday, previousMonday, startOfDay } from "date-fns";
+import { FontAwesomeSvgIcon } from 'react-fontawesome-svg-icon';
+import { faPlus } from '@fortawesome/free-solid-svg-icons';
+import { BloodPressureDataParameters, BloodPressureDataPoint, BloodPressureDataProvider } from "../../../helpers/blood-pressure-data-providers/query-blood-pressure";
+
+export enum Category {"Low", "Normal", "Elevated", "Stage 1", "Stage 2", "Crisis", "Unknown"};
+export type BloodPressurePreviewState = "WithData" | "NoData" | "Loading" | "Live";
+
+export interface BloodPressureVisualizationProps {
+    previewState? : BloodPressurePreviewState,
+    previewDataProvider : BloodPressureDataProvider,
+    dataParameters : BloodPressureDataParameters
+};
+
+
+interface BloodPressureDumbbell {
+    date: Date,
+    dumbbell : Dumbbell,
+    averageSystolic : number,
+    averageDiastolic : number
+};
+
+interface BloodPressureMetrics {
+    averageSystolic? : number,
+    averageSystolicAlert? : string,
+    averageSystolicAlertClass : string,
+    averageDiastolic? : number,
+    averageDiastolicAlert? : string,
+    averageDiastolicAlertClass : string,
+    maxSystolic? : number,
+    maxSystolicAlert? : string,
+    maxSystolicAlertClass : string,
+    minDiastolic? : number, 
+    minDiastolicAlert? : string,
+    minDiastolicAlertClass : string
+}
+
+export default function (props : BloodPressureVisualizationProps) {
+    const _minDiastolic = 0;
+    const _maxSystolic = 250; 
+    const yInterval : ClosedInterval = {values: [_minDiastolic, _maxSystolic]};
+    const axis : Axis = {yRange : yInterval, yIncrement : 50, xIncrement : (85/7)};
+
+    const [loadingData, setLoadingData] = useState(false);
+    const [bpData, setBpData] = useState<BloodPressureDumbbell[]>([]);
```

Comment:
So thinking about what you actually need for state...you don't need to be holding onto this much state.

I would have 2 state variables:

```
const [bloodPressureData, setBloodPressureData] = useState<BloodPressureDataPoint[] | null>(null);
const [startOfWeek, setStartOfWeek] = useState<Date>(initializeWeekPager());
```

You can compute everything else based on those 2 state variables.

Reply:
(Loading can be inferred by a null bloodPressureData)

Reply:
I am having a tough time getting the weeklyData and weeklyMetrics out of state.

Reply:
I would think about it this way...you should be able to create a separate standalone method which takes in 2 parameters (startOfWeek and bloodPressureData) - and it should be able to return weeklyData/weeklyMetrics.  Unless there is some other dependency i'm overlooking?

Reply:
I was able to reduce like suggested. Thank you for this feedback